src/dma/traits.rs

```diff
@@ -0,0 +1,1394 @@
+    (Stream7<DMA1>, Channel3, DMAR<pac::TIM2>, PeripheralToMemory), //TIM2_UP
+);
+
+#[cfg(any(feature = "stm32f401", feature = "stm32f411",))]
+address!((CCR3<pac::TIM2>, ccr3, u16), (DMAR<pac::TIM2>, dmar, u16),);
+
+#[cfg(any(
+    feature = "stm32f401",
+    feature = "stm32f411",
+    feature = "stm32f412",
+    feature = "stm32f413",
+    feature = "stm32f423",
+))]
+dma_map!((Stream5<DMA1>, Channel6, pac::I2C3, MemoryToPeripheral),); //I2C3_TX:DMA_CHANNEL_6);
+
+#[cfg(any(
+    feature = "stm32f401",
+    feature = "stm32f417",
+    feature = "stm32f415",
+    feature = "stm32f405",
+    feature = "stm32f407",
+    feature = "stm32f427",
+    feature = "stm32f439",
+    feature = "stm32f437",
+    feature = "stm32f429",
+    feature = "stm32f446",
+    feature = "stm32f469",
+    feature = "stm32f479",
+))]
+dma_map!(
+    (Stream6<DMA1>, Channel1, pac::I2C1, MemoryToPeripheral), //I2C1_TX
+    (Stream7<DMA1>, Channel1, pac::I2C1, MemoryToPeripheral), //I2C1_TX
+    (Stream3<DMA2>, Channel3, pac::SPI1, MemoryToPeripheral), //SPI1_TX
+    (Stream5<DMA2>, Channel3, pac::SPI1, MemoryToPeripheral), //SPI1_TX
+);
+
+#[cfg(any(
+    feature = "stm32f401",
+    feature = "stm32f411",
+    feature = "stm32f412",
+    feature = "stm32f413",
+    feature = "stm32f423",
+    feature = "stm32f427",
+    feature = "stm32f439",
+    feature = "stm32f437",
+    feature = "stm32f429",
+    feature = "stm32f446",
+    feature = "stm32f469",
+    feature = "stm32f479",
+))]
+dma_map!(
+    (Stream0<DMA2>, Channel4, pac::SPI4, PeripheralToMemory), //SPI4_RX
+    (Stream1<DMA2>, Channel4, pac::SPI4, MemoryToPeripheral), //SPI4_TX
+    (Stream3<DMA2>, Channel5, pac::SPI4, PeripheralToMemory), //SPI4_RX:DMA_CHANNEL_5
+    (Stream4<DMA2>, Channel5, pac::SPI4, MemoryToPeripheral), //SPI4_TX:DMA_CHANNEL_5
+);
+
+#[cfg(any(
+    feature = "stm32f401",
+    feature = "stm32f411",
+    feature = "stm32f412",
+    feature = "stm32f413",
+    feature = "stm32f423",
+    feature = "stm32f427",
+    feature = "stm32f439",
+    feature = "stm32f437",
+    feature = "stm32f429",
+    feature = "stm32f446",
+    feature = "stm32f469",
+    feature = "stm32f479",
+))]
+address!((pac::SPI4, dr, u8),);
+
+#[cfg(any(
+    feature = "stm32f417",
+    feature = "stm32f415",
+    feature = "stm32f405",
+    feature = "stm32f407",
+    feature = "stm32f413",
+    feature = "stm32f423",
+    feature = "stm32f427",
+    feature = "stm32f439",
+    feature = "stm32f437",
+    feature = "stm32f429",
+    feature = "stm32f446",
+    feature = "stm32f469",
+    feature = "stm32f479",
+))]
+dma_map!(
+    (Stream0<DMA1>, Channel4, pac::UART5, PeripheralToMemory), //UART5_RX
+    (Stream2<DMA1>, Channel4, pac::UART4, PeripheralToMemory), //UART4_RX
+    (Stream4<DMA1>, Channel4, pac::UART4, MemoryToPeripheral), //UART4_TX
+                                                               //(pac::DMA1, Stream6, Channel7, pac::DAC2, MemoryToPeripheral), //DAC2
```

Comment:
I looked at some of those, it seems that it needs more work in the peripheral itself (new types, etc) to be able to conform with our API here, I left some `TODO`s around, I remember putting a TODO for the DAC somewhere, hmmm.

Reply:
Yes, most of the commented out code already has comments saying why it's commented out (mostly just because there seem to be registers missing in the PAC?). Looks like there is just one missing here.